### ISSUE TITLE
Display hazard nodes

### ISSUE THREAD

**Issue by troelsarvin** [osmandapp/OsmAnd#16810]
# 🚀 feature request

### Description

[I've tagged a rather dangerous place in Turkey](https://troels.arvin.dk/osm/topics/hole_in_ground_in_tyrkey/), using the using the "hazard" tag. When tagging, I asked for advise in the mailing list on how to do it, and the current tagging is evidently the correct way to tag things.

Unfortunately, OsmAnd doesn't display the hazard, see attached screenshot below.
OsmAnd does, however, display the other tag I added, displaying an icon for ruins.
![sshot_osmand_no_hazard_indication](https://user-images.githubusercontent.com/5257280/227782346-28e96db9-c9f8-45f2-8406-83d675ad14c0.jpg)

End result is that my efforts may now be luring people to the area, bringing themselves in harms way, because they may fall down the hole which isn't being warned about. I'm uneasy about that.

### Suggested solution

I suggest OsmAnd display a hazard-icon for nodes which has the "hazard" tag. And maybe even sound an alert, sort of like OsmAnd does when you are (for example) approaching a speed bump.

**Comment by Zirochkabila** [comment]
[TagInfo hazard=hole](https://taginfo.openstreetmap.org/search?q=hazard%3Dhole+)
[TagInfo hazard](https://taginfo.openstreetmap.org/search?q=hazard)
[OSM](https://www.openstreetmap.org/way/1126235614)


@troelsarvin  The offer is interesting
But I'm not sure we'll implement it because of the small number of tags
We need to clarify with @vshcherb and @xmd5a2

**Comment by scaidermern** [comment]
It is not strictly necessary to support hazard=hole in OsmAnd. It would be sufficient to add a generic rendering for any unknown hazard=* tag.

**Comment by Max1234-Ita** [comment]
> # 🚀 feature request
> ### Description
> 
> [I've tagged a rather dangerous place in Turkey](https://troels.arvin.dk/osm/topics/hole_in_ground_in_tyrkey/), using the using the "hazard" tag. When tagging, I asked for advise in the mailing list on how to do it, and the current tagging is evidently the correct way to tag things.
> 
> Unfortunately, OsmAnd doesn't display the hazard, see attached screenshot below. OsmAnd does, however, display the other tag I added, displaying an icon for ruins. ![sshot_osmand_no_hazard_indication](https://user-images.githubusercontent.com/5257280/227782346-28e96db9-c9f8-45f2-8406-83d675ad14c0.jpg)
> 
> End result is that my efforts may now be luring people to the area, bringing themselves in harms way, because they may fall down the hole which isn't being warned about. I'm uneasy about that.
> ### Suggested solution
> 
> I suggest OsmAnd display a hazard-icon for nodes which has the "hazard" tag. And maybe even sound an alert, sort of like OsmAnd does when you are (for example) approaching a speed bump.


Such a feature might also be useful to warn drivers about particular spots, i.e. dangerous bends/crossings: that would really improve road safety.

A long time ago, before OsmAnd was created, I had a Tomtom navigator: I had found an additional POI category (created by a community of users) where a lot of such points were collected. 
It was a real help having that category active while driving around, and hearing an alert when approaching to one of them: I'm definitely missing that feature in OsmAnd.

**Comment by vshcherb** [comment]
Yes we can display hazard nodes and I think we have icon for that

**Comment by xmd5a2** [comment]
Dev notes: for some reason previous [commit](https://github.com/osmandapp/OsmAnd-resources/commit/dd9589e59603fcf41e4c6c8a6f61ebf59cd9f5ed) has no effect. Issue is related to tiny polygon area https://www.openstreetmap.org/way/1126235614#map=19/36.360817/30.476439 . For some reason it is hidden in legacy and OpenGL engines. If I enlarge it manually then it renders correctly. Map data are correct.
Rendering of manually enlarged area:

![Image](https://github.com/user-attachments/assets/36ae234d-4eb1-40cb-bcaa-4edbcc1c4560)

**Comment by ihoryakovenko** [comment]
The problem is not in the engine, but in the OBF generation. I used a small polygon approximately 10x10 meters in 0.0, 0.0 coordinates for testing. 
[hazard_small.osm.zip](https://github.com/user-attachments/files/20630234/hazard_small.osm.zip)
After obf generation, BinaryMapObject contains 0 points. This can be seen in JOSM after converting the OBF back to OSM.
[hazard_small_obf.osm.zip](https://github.com/user-attachments/files/20630406/hazard_small_obf.osm.zip)

**Comment by xmd5a2** [comment]
@troelsarvin We don't plan to support such a tiny polygons as icons. This may result in cluttering of the map. Please remap this object as a **node**.

**Comment by scaidermern** [comment]
> Please remap this object as a **node**.

Maybe OsmAnd should simplify it to a node during map generation.

**Comment by Max1234-Ita** [comment]
> > Please remap this object as a **node**.
> 
> Maybe OsmAnd should simplify it to a node during map generation.

I don't think it's a viable solution, it would require a lot of additional checks on all the data being imported, in order to detect those objects that are "small enough" to be represented as a single node.

OsmAnd uses OSM data, its purpose is not to "correct" them automatically, IMO.

**Comment by scaidermern** [comment]
> OsmAnd uses OSM data, its purpose is not to "correct" them automatically, IMO.

Well, so far there is no consensus whether this is wrong data so I wouldn't call it correcting it.

The previous suggestion was to change this OSM data just because one particular data consumer has a problem with it. This is obviously not a good solution.

**Comment by RZR-UA** [comment]
This object is mapped with an area less than 1 m2. It would be better to remap it to a node.

![Image](https://github.com/user-attachments/assets/b61e9cee-67a0-4a1f-9231-5c14b7250027)

**Comment by troelsarvin** [comment]
I've mapped the hazard as a node now. It would certainly in general be nice if OsmAnd could display very small areas, but I agree it's time to close this issue.

**Comment by Zirochkabila** [comment]
# Testing 1
`OsmAnd~ 5.1.0#5381m, Released: 2025-06-13`
`5.0.5.5060`

`hazard=hole` d OsmAnd tag is converted to `hazard=other` - **Added**


<details><summary>Details</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/93dc985f-f6dc-4981-94c2-9df3bed218f0"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/326b2330-67b0-4148-8636-9a487c300c71"  width="300" height="auto"/> </th>
  </tr>
</table>  
</details>
